### PR TITLE
Update pager template to use blocks

### PR DESCRIPTION
**Pull request by CarsonF**
This abstracts the logic around creating a pager into simple blocks.

See the [pager/default.twig](https://github.com/CarsonF/bolt/blob/feature/pager-blocks/app/theme_defaults/pager/default.twig) for an example.

### CODE REVIEW

**Comment by bobdenotter** [comment]
Nice! I'd just like to add some html/classes to some items, for semantics and so that they can be styled.

File: src/Storage/Field/Collection/FieldCollection.php [hunk 1]
```diff
@@ -146,16 +146,6 @@ public function offsetGet($offset)
     }
 
     /**
-     * Overrides the default array access method and passes on to the get method
-     * @param mixed $offset
-     * @return mixed
-     */
-    public function offsetGet($offset)
```
Comment (rossriley) [inline src/Storage/Field/Collection/FieldCollection.php]: Should this be part of the PR?

Comment (CarsonF) [inline src/Storage/Field/Collection/FieldCollection.php]: It's from a previous merge conflict. It could be committed separately

**Comment by CarsonF** [comment]
@bob can you expand on that?

File: src/Storage/Field/Collection/FieldCollection.php (hunk 1 above)
Comment (GawainLynch) [inline src/Storage/Field/Collection/FieldCollection.php]: @rossriley what @CarsonF said … it was my bad in the first place, for bigger branch merges I usually run tests before committing the merge, but you know that time you drop process 'cause "It'll be fiiiiiiiiiiiiiiiiiiiine!"

Admittedly it isn't best practice and I can move to a new PR in a few hours when I get up for real. My motivation is that @CarsonF is away for the week, and my time zones sucks too coordinate merges, and I was super excited about this PR. 

Not meant to excuse, just an explanation :smile:

**Comment by rossriley** [comment]
No worries. Just checking it wasn't a mistake. I've actually just run into
the fatal so let's get it merged.

**Comment by GawainLynch** [comment]
Sure … Opened a #6383 so that we can follow up on the request by @bobdenotter

**Pull request merged**